source/Preprocessor.cpp

```diff
@@ -844,7 +844,23 @@ MacroActualArgumentSyntax* Preprocessor::MacroParser::parseActualArgument() {
 }
 
 MacroFormalArgumentSyntax* Preprocessor::MacroParser::parseFormalArgument() {
-    auto arg = expect(TokenKind::Identifier);
+    Token arg;
```

Comment:
This seems like a complicated way to do this. Could we instead peek() the current token; if it's a keyword or an identifier, consume() it. Otherwise call expect(TokenKind::Identifier) to have it fabricate a placeholder for us.